ignition_ros2_control/src/ignition_ros2_control_plugin.cpp

```diff
@@ -0,0 +1,400 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <map>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "controller_manager/controller_manager.hpp"
+
+#include "hardware_interface/resource_manager.hpp"
+#include "hardware_interface/component_parser.hpp"
+#include "hardware_interface/types/hardware_interface_type_values.hpp"
+
+
+#include "ignition_ros2_control/ignition_ros2_control_plugin.hpp"
+#include "ignition_ros2_control/ignition_system.hpp"
+
+#include "ignition/gazebo/components/Joint.hh"
+#include "ignition/gazebo/components/JointType.hh"
+#include "ignition/gazebo/components/Name.hh"
+#include "ignition/gazebo/components/ParentEntity.hh"
+#include "ignition/gazebo/components/Physics.hh"
+#include "ignition/gazebo/components/World.hh"
+#include "ignition/gazebo/Model.hh"
+
+#include "ignition/plugin/Register.hh"
+
+#include "pluginlib/class_loader.hpp"
+
+#include "rclcpp/rclcpp.hpp"
+
+namespace ignition_ros2_control
+{
+//////////////////////////////////////////////////
+class IgnitionROS2ControlPluginPrivate
+{
+  /// \brief Entity ID for sensor within Gazebo.
+
+public:
+  ignition::gazebo::Entity entity_;
+
+public:
+  std::shared_ptr<rclcpp::Node> node;
+
+  // Thread where the executor will sping
+  std::thread thread_executor_spin_;
+
+  // Executor to spin the controller
+  rclcpp::executors::MultiThreadedExecutor::SharedPtr executor_;
+
+  // Timing
+  rclcpp::Duration control_period_ = rclcpp::Duration(1, 0);
+
+  // Interface loader
+  boost::shared_ptr<pluginlib::ClassLoader<
+      ignition_ros2_control::IgnitionSystemInterface>> robot_hw_sim_loader_;
+
+  // Controller manager
+  std::shared_ptr<controller_manager::ControllerManager> controller_manager_;
+
+  // String with the robot description
+  // TODO(ahcorde): Add param in plugin tag
+  std::string robot_description_ = "robot_description";
+  // String with the name of the node that contains the robot_description
+  // TODO(ahcorde): Add param in plugin tag
+  std::string robot_description_node_ = "robot_state_publisher";
+
+  // Last time the update method was called
+  rclcpp::Time last_update_sim_time_ros_;
+
+  std::string getURDF(std::string param_name) const
```

Comment:
Please document this function. Also consider separating the function declaration from the implementation so it's a bit more organized.

Also, I think it's good to be consistent with the capitalization. The other functions are `CamelCased`.